src/libcore/num/mod.rs

```diff
@@ -589,6 +592,21 @@ macro_rules! int_impl {
                     v   => Some(self / v),
                 }
             }
+
+            #[inline]
+            fn pow(self, mut exp: u32) -> Self {
+                let mut base = self;
+                let mut acc: Self = Int::one();
+
+                while exp > 0 {
+                    if (exp & 1) == 1 {
+                        acc = acc * base;
+                    }
+                    base = base * base;
+                    exp /= 2;
+                }
+                acc
+            }
```

Comment:
Would it be possible to leave the default method implementation for these methods? It looks like both impls are the same currently.